lib/cloudregister/registerutils.py

```diff
@@ -101,13 +107,21 @@ def clean_hosts_file(domain_name):
         content = hosts_file.readlines()
 
     smt_announce_found = None
+    smt_domain_found = None
     for entry in content:
         if b'# Added by SMT' in entry:
             smt_announce_found = True
             continue
         if smt_announce_found and domain_name in entry:
             smt_announce_found = False
+            smt_domain_found = True
             continue
+        if smt_domain_found:
+            smt_domain_found = False
```

Comment:
This makes the assumption that the registry entry is right after the repository entry. Which is what we create. However, users do stuff we don't want them to do. Therefore the flag needs to be set to `False` after we identify the line we want to remove. This means that we inspect every line after the announce flag is found if there is no registry entry.

For example
```
# Added by SMT registration do not remove, retain comment as well
2.2.2.2 smt-ec2.susecloud.net smt-ec2
1.1.1.1 my.specialhost.us
2.2.2.2 registry-ec2.susecloud.net smt-ec2
```
In this case the registry entry would not get cleaned up.